base/uk.ac.stfc.isis.ibex.e4.ui.perspectiveswitcher/src/uk/ac/stfc/isis/ibex/e4/ui/perspectiveswitcher/controls/Button.java

```diff
@@ -42,9 +44,23 @@ public Button(Composite parent, String imageUri, String tooltip, ButtonViewModel
         super(parent, SWT.SHADOW_OUT);
 
         setLayout(new GridLayout());
-        GridData gridData = new GridData(SWT.FILL, SWT.FILL, true, true);
-        // Needs to be wider than default to allow for larger text because its font is bold when the button is active.
-        gridData.minimumWidth = buttonWidthMin;
+        setRightMargin(0);
+        
+        GridData gridData = new GridData(SWT.LEFT, SWT.CENTER, false, false);
+        model.addPropertyChangeListener("width", new PropertyChangeListener() {
+            
+            @Override
+            public void propertyChange(PropertyChangeEvent evt) {
+                Display.getDefault().asyncExec(new Runnable() {
+                    
+                    @Override
+                    public void run() {
+                        gridData.widthHint = model.getWidth();
```

Comment:
If you use `(int) evt.getNewValue()` then the model will not have to store the width.